### PR TITLE
reload component if given list of program years changes.

### PR DESCRIPTION
**Pull request by stopfstedt**
fixes https://github.com/ilios/frontend/issues/5886

### CODE REVIEW

**Comment by stopfstedt** [comment]
_side note:_ we need more/better test coverage surrounding program-year creation.  refs https://github.com/ilios/frontend/issues/5521

File: app/components/new-programyear.js [hunk 1]
```diff
@@ -4,6 +4,7 @@ import { action } from '@ember/object';
 import { dropTask } from 'ember-concurrency';
 
 export default class NewProgramyearComponent extends Component {
+  @tracked allYears;
```
Comment (jrjohnson) [inline app/components/new-programyear.js:7]: Doesn't matter in any way, but since this value never changes it doesn't actually need to be `@tracked`.

**Review by jrjohnson — APPROVED** [review]

File: app/components/new-programyear.js (hunk 1 above)
Comment (stopfstedt) [reply app/components/new-programyear.js:7]: :heavy_check_mark:

**Review by dartajax — APPROVED** [review]
It looks good to me.

**Pull request merged**